src/components/_button.scss

```diff
@@ -59,210 +59,145 @@ $btn-padding-x: (
-    border-bottom-color: colors.$success-dark;
-
-    &:hover,
-    &:focus {
-      background-color: colors.$success-hover;
-    }
-
-    &:focus {
-      box-shadow: 0px 0px 0px 3px rgba(73, 201, 109, 0.3);
-    }
-
-    &:active {
-      background-color: colors.$success-dark;
-    }
-  }
-
-  &.light {
-    background-color: #fff;
-    border-bottom-color: transparent;
-
-    &:hover {
-      background-color: colors.$bg-dark;
-    }
-
-    &.primary {
-      background-color: colors.$primary-light;
-      color: colors.$primary;
+  @mixin button(
+    $color,
+    $bg: colors.$primary,
+    $hover: colors.$primary-hover,
+    $active: colors.$primary-dark,
+    $focus: colors.$primary-shadow
+  ) {
+    &.#{$color} {
+      background-color: $bg;
+      color: #fff;
+      border-bottom-color: $active;
 
       &:hover,
       &:focus {
-        background-color: colors.$primary;
-        color: #fff;
+        background-color: $hover;
       }
 
       &:focus {
-        box-shadow: 0px 0px 0px 3px rgba(48, 79, 253, 0.3);
+        box-shadow: 0px 0px 0px 3px $focus;
       }
 
       &:active {
-        background-color: colors.$primary-dark;
+        background-color: $active;
       }
     }
+  }
 
-    &.warning {
-      background-color: colors.$warning-light;
-      color: colors.$warning;
-
-      &:hover,
-      &:focus {
-        background-color: colors.$warning;
-        color: #fff;
-      }
-
-      &:focus {
-        box-shadow: 0px 0px 0px 3px rgba(229, 110, 25, 0.3);
-      }
-
-      &:active {
-        background-color: colors.$warning-dark;
-      }
-    }
-
-    &.danger {
-      background-color: colors.$danger-light;
-      color: colors.$danger;
-
-      &:hover,
-      &:focus {
-        background-color: colors.$danger;
-        color: #fff;
-      }
+  @include button(primary);
+
+  @include button(
+    warning,
+    colors.$warning,
+    colors.$warning-hover,
+    colors.$warning-dark,
+    colors.$warning-shadow
+  );
+
+  @include button(
+    danger,
+    colors.$danger,
+    colors.$danger-hover,
+    colors.$danger-dark,
+    colors.$danger-shadow
+  );
+
+  @include button(
+    success,
+    colors.$success,
+    colors.$success-hover,
+    colors.$success-dark,
+    colors.$success-shadow
+  );
 
-      &:focus {
-        box-shadow: 0px 0px 0px 3px rgba(253, 121, 114, 0.3);
-      }
+  &.light {
+    background-color: #fff;
+    border-bottom-color: transparent;
 
-      &:active {
-        background-color: colors.$danger-dark;
-      }
+    &:hover {
+      background-color: colors.$bg-dark;
     }
 
-    &.success {
-      background-color: colors.$success-light;
-      color: colors.$success;
-
-      &:hover,
-      &:focus {
-        background-color: colors.$success;
-        color: #fff;
-      }
-
-      &:focus {
-        box-shadow: 0px 0px 0px 3px rgba(73, 201, 109, 0.3);
-      }
-
-      &:active {
-        background-color: colors.$success-dark;
-      }
-    }
+    @mixin light(
```

Comment:
Why this mixin's content cannot be included in the above one? I want to call a single mixin that generates ALL variations of a button